### PR TITLE
Use dedicated exception for scalar type parsing error

### PR DESCRIPTION
**Pull request by mcg-web**
see #123

### CODE REVIEW

File: src/Error/UserError.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+/**
+ * Class UserError
+ *
+ * Note:
+ * Error that can be display safely to client...
+ *
+ * @package GraphQL\Error
+ */
+class UserError extends InvariantViolation
```
Comment (mcg-web) [inline src/Error/UserError.php]: extends `InvariantViolation` to avoid BC, next major version we can change this.

Comment (TerraSkye) [inline src/Error/UserError.php]: I think its an acceptable BC, if we put in release notes, as it is a security enhanchement?

Comment (mcg-web) [inline src/Error/UserError.php]: BC is never acceptable or not it is just a reality, if your new release reflect the BC no trouble but it is very important to dependencies...

Comment (TerraSkye) [inline src/Error/UserError.php]: Yeah i forgot that part. My bad, your suggestion seems the best then!

File: src/Type/Definition/ObjectType.php [hunk 2]
```diff
@@ -145,7 +145,7 @@ public function getInterfaces()
             foreach ($interfaces as $iface) {
                 $iface = Type::resolve($iface);
                 if (!$iface instanceof InterfaceType) {
-                    throw new InvariantViolation("Expecting interface type, got " . Utils::printSafe($iface));
+                    throw new UserError(sprintf('Expecting interface type, got %s', Utils::printSafe($iface)));
```
Comment (vladar) [inline src/Type/Definition/ObjectType.php]: This looks like a perfect InvariantViolationError for me, as it's server developer error (in type definition). Is there any reason why this must be UserError?

Comment (mcg-web) [inline src/Type/Definition/ObjectType.php]: ok i'm rolling back this one!

**Pull request merged**